calcom/cal.com
# fix: Handle payment flow webhooks in case of event requiring confirmation

## What does this PR do?

Fixes #7120

- Adds BOOKING_PAYMENT_INITIATED webhook
- Fixes the issue where, when a paid booking that require confirmation was submitted
	-  BOOKING_REQUESTED webhook didn't fire
	-  Booking Request related emails weren't sent to the organizer and booker.
- Also, correctly takes BOOKING_CREATED/BOOKING_REQUESTED flow for an event that requires confirmation based on duration notice threshold

Also, adds tests in that flow and few other booking flows.
Also, added tests for the stripe webhook flow which complements the booking flow.

<img width="1191" alt="Screenshot 2023-09-06 at 4 54 51 PM" src="https://github.com/calcom/cal.com/assets/1780212/3ad263fa-3ad6-4f3f-acc1-3acddf9aad30">

There is a small refactor, cleanup and addition of more mocks in **bookingScenario** as well
- Made it easy to write tests by adding more reusable expects
- Mocked stripe payment flow

## Type of change

## How should this be tested?
You will need to setup stripe related env variables
- [x] Test regular event
	- Create an event type that requires payment. Also, enables requires confirmation for the event type.
	- Do a booking for that event type, reach the payment page. Expect an email to booker at the moment asking for completing the payment.
		- Also verify at this moment that BOOKING_PAYMENT_INITIATED webhook is fired. _This is new_
	- Complete the payment. Expect 2 more emails. One to booker that booking has been submited and other to organizer that booking is awaiting approval. _This wasn't working earlier. This has been fixed._
		- Also verify that BOOKING_REQUESTED webhook is fired.
- [x] Test the same for a Team Event - (Tested with collective)
- [x] Test the same for Seated Event
- [x] Test same Managed Event Type
	- Notable Observation: 2 Webhooks fires for each event when both team and user webhook are created.

## Mandatory Tasks
